index.ts

```diff
@@ -28,6 +29,126 @@ import {
+ */
+async function fetchBlockData(proposal: any, latestBlock: any) {
+  const [startBlock, endBlock] = await Promise.all([
+    proposal.startBlock <= (latestBlock.number ?? 0n)
+      ? publicClient.getBlock({ blockNumber: proposal.startBlock })
+      : null,
+    proposal.endBlock <= (latestBlock.number ?? 0n)
+      ? publicClient.getBlock({ blockNumber: proposal.endBlock })
+      : null,
+  ]);
+
+  return {
+    current: latestBlock,
+    start: startBlock,
+    end: endBlock,
+  };
+}
+
+/**
+ * @notice Process cross-chain destination simulations and run checks
+ */
+async function processDestinationSimulations(
+  proposal: any,
+  deps: ProposalData,
+  destinationSimulations: SimulationResult['destinationSimulations'],
+) {
+  const destinationChecks: Record<number, AllCheckResults> = {};
+
+  if (destinationSimulations) {
+    for (const destSim of destinationSimulations) {
+      if (destSim.sim) {
+        const l2Deps = {
+          ...deps,
+          chainConfig: getChainConfig(destSim.chainId),
+        };
+        destinationChecks[destSim.chainId] = await runChecksForChain(
+          proposal,
+          destSim.sim,
+          l2Deps,
+          destSim.chainId,
+          destinationSimulations,
+        );
+      }
+    }
+  }
+
+  return destinationChecks;
+}
+
+
+
+/**
+ * @notice Process a single simulation with checks and reporting
+ */
+async function processSimulation(
+  config: SimulationConfig,
+  governorType: GovernorType,
+  proposalData: ProposalData,
+  simulationResult: SimulationResult,
+  proposalId: string,
+  proposalState: string,
+) {
+  const { sim, proposal, latestBlock, proposalCreatedBlock, proposalExecutedBlock, executor } = simulationResult;
+
+  // Run checks
+  console.log(`  Running checks for proposal ${proposalId}...`);
+  const checkResults: AllCheckResults = Object.fromEntries(
+    await Promise.all(
+      Object.keys(ALL_CHECKS).map(async (checkId) => [
+        checkId,
+        {
+          name: ALL_CHECKS[checkId].name,
+          result: await ALL_CHECKS[checkId].checkProposal(proposal, sim, proposalData),
+        },
+      ]),
+    ),
+  );
+
+  // Fetch block data
+  const blocks = await fetchBlockData(proposal, latestBlock);
+
+  // Generate reports
+  const dir = `./reports/${config.daoName}/${config.governorAddress}`;
+  await generateAndSaveReports({
+    governorType,
+    blocks,
+    proposal,
+    checks: checkResults,
+    outputDir: dir,
+    governorAddress: config.governorAddress,
+    executor,
+    proposalCreatedBlock,
+    proposalExecutedBlock,
+  });
+
+  // Cache results
+  const simulationData: SimulationData = {
+    sim,
+    proposal,
+    latestBlock,
+    config,
+    deps: proposalData,
+    proposalCreatedBlock,
+    proposalExecutedBlock,
+    executor,
+  };
+
+  await cacheProposal(
+    config.daoName,
+    config.governorAddress,
+    proposal.id.toString(),
+    proposalState, // ASK: MARCO : THIS WAS HARCODED TO 1
```

Comment:
Just curious if we can remove the harcoding

Reply:
yeee! nice find! i think this is why the caching isnt working reliably and i was trying to fix it here:

https://github.com/uniswapfoundation/governance-seatbelt/pull/43

let's see if this fixes the issue in the gov checks in ci and i'll close the above pr.